src/spyglass/utils/dj_helper_fn.py

```diff
@@ -289,3 +307,159 @@ def get_child_tables(table):
+                f"Adjusting subject sex from '{sex_value}' to '{new_sex_value}'."
+            )
+            f["/general/subject/sex"][()] = new_sex_value
+        elif sex_value == "Female":
+            new_sex_value = "F"
+            print(
+                f"Adjusting subject sex from '{sex_value}' to '{new_sex_value}'."
+            )
+            f["/general/subject/sex"][()] = new_sex_value
+
+        # replace subject species value "Rat" with "Rattus norvegicus"
+        species_value = f["/general/subject/species"][()].decode("utf-8")
+        assert species_value in ("Rat", "Rattus norvegicus")
+        if species_value == "Rat":
+            new_species_value = "Rattus norvegicus"
+            print(
+                f"Adjusting subject species from '{species_value}' to '{new_species_value}'."
+            )
+            f["/general/subject/species"][()] = new_species_value
+
+        # add subject age dataset "P4M/P8M"
+        if "age" not in f["/general/subject"]:
+            new_age_value = age
+            print(f"Adding missing subject age, set to '{new_age_value}'.")
+            f["/general/subject"].create_dataset(
+                name="age", data=new_age_value, dtype=STR_DTYPE
+            )
+
+        # format name to "Last, First"
+        experimenter_value = f["/general/experimenter"][:].astype(str)
+        new_experimenter_value = dandi_format_names(experimenter_value)
+        if experimenter_value != new_experimenter_value:
+            new_experimenter_value = new_experimenter_value.astype(STR_DTYPE)
+            print(
+                f"Adjusting experimenter from {experimenter_value} to {new_experimenter_value}."
+            )
+            f["/general/experimenter"][:] = new_experimenter_value
+
+    # update the datajoint external store table to reflect the changes
+    _resolve_external_table(filepath, file_name)
+
+
+def dandi_format_names(experimenter: List) -> List:
+    """Make names compliant with dandi standard of "Last, First"
+
+    Parameters
+    ----------
+    experimenter : List
+        List of experimenter names
+
+    Returns
+    -------
+    List
+        reformatted list of experimenter names
+    """
+    for i, name in enumerate(experimenter):
+        parts = name.split(" ")
+        new_name = " ".join(
+            parts[:-1],
+        )
+        new_name = f"{parts[-1]}, {new_name}"
+        experimenter[i] = new_name
+    return experimenter
+
+
+def _resolve_external_table(
+    filepath: str, file_name: str, location: str = "analysis"
+):
+    """Function to resolve discrepancies between the database external checks and
+    current file properties.
+    WARNING: This should only be used when editing file metadata. Can violate data
+    integrity if impproperly used.
+
+    Parameters
+    ----------
+    filepath : _type_
+        _description_
+    file_name : _type_
+        _description_
+    location : str, optional
+        _description_, by default "analysis"
```

Comment:
Looks like this has some defaults still in place. copilot does a lot of docstring writing for me